src/lang/checkers/Cashflow.ml

```diff
@@ -209,14 +219,20 @@ module ScillaCashflowChecker
       CFSyntax.tbody =
         List.map ~f:cf_init_tag_stmt tbody }
   
-  let cf_init_tag_contract contract =
+  let cf_init_tag_contract contract token_fields =
     let { cname ; cparams ; cfields ; ctrans } = contract in
     { CFSyntax.cname = cname;
       CFSyntax.cparams =
-        List.map ~f:(fun (x, t) -> (add_noinfo_to_ident x, t)) cparams;
+        List.map ~f:(fun (x, t) ->
+            (if List.exists ~f:(fun token_field -> get_id x = token_field) token_fields
+             then add_money_or_mapmoney_to_ident x t
+             else add_noinfo_to_ident x),
+            t) cparams;
       CFSyntax.cfields =
         List.map ~f:(fun (x, t, e) ->
-            (add_noinfo_to_ident x,
+            ((if List.exists ~f:(fun token_field -> get_id x = token_field) token_fields
```

Comment:
This snippet could probably be made into a local definition as it repeats.

Reply:
Fixed.